geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientUpdateMessageImpl.java

```diff
@@ -1011,7 +1015,7 @@ public void sendTo(DataOutput out) throws IOException {
       InternalDataSerializer.writeArrayLength(size, out);
       if (size > 0) {
         DataSerializer.writeObject(name[0], out);
-        DataSerializer.writeObject(op, out);
+        DataSerializer.writeObject(op.id, out);
```

Comment:
Is there a possibility of a NPE here or other places if we call members of op, if we created CqNameToOpSingleEntry with op value as null. As we can see is possible in by the new code change in HAEventWrapper.

Reply:
It looks like no, because when constructed with `null` the `name` will also be `null`. This means that if a message is added then the op is taking from the first message added, otherwise if it remains empty when serialized the `op.id` is never referenced. To be absolutely clear though I am going to `@NotNull` in that constructor, add a new default constructor for the empty case so it isn't called with `null`, guaranteeing that both `name` and `op` will be `null` together. Good find!!

Reply:
Actually on further inspection it is all sort of wonky and probably does lead to a NPE. There seems to be a break in the original logic around what makes it `isEmpty()`. If constructed with a `null` name the `isEmpty()` did not return `true` as it would be expected to. I am going to whip up some unit tests around this class and correct. Clearly there aren't any tests that hit this scenario either or at least none that fail. In the original code you would have had `null` and `0` serialized on the wire for an empty entry like this. `0` meaning it was a `REQUEST` message. Invoking the `add()` method would have resulted in the NPE. 

Great find!

Reply:
Ok, I refactored the commits, added some tests around this untested class in question and fixed the bug pointed out here.